ledger/db/src/test_utils/mock_ledger.rs

```diff
@@ -300,11 +301,139 @@ pub fn get_test_ledger_blocks(n_blocks: usize) -> Vec<(Block, BlockContents)> {
+    key_images_per_block: usize,
+    max_token_id: u64,
+) -> Vec<(Block, BlockContents)> {
+    let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+
+    // Number of total tx outputs in all blocks
+    let num_outputs: u64 =
+        (num_accounts * outputs_per_recipient_per_block * num_blocks * (max_token_id as usize + 1))
+            as u64;
+    assert!(num_outputs >= 16);
+
+    // Initialize other defaults
+    let picomob_per_output: u64 = (TOTAL_MOB / num_outputs) * 1_000_000_000_000;
+    let recipients = (0..num_accounts)
+        .map(|_| AccountKey::random(&mut rng).default_subaddress())
+        .collect::<Vec<_>>();
+    let block_version = BlockVersion::MAX;
+    let mut blocks_and_contents: Vec<(Block, BlockContents)> = Vec::new();
+    let mut previous_block: Option<Block> = None;
+
+    // Create the tx outs for all of the simulated blocks
+    for _ in 0..num_blocks as u64 {
+        let mut outputs: Vec<TxOut> = Vec::new();
+        for recipient in &recipients {
+            let tx_private_key = RistrettoPrivate::from_random(&mut rng);
+            for _ in 0..outputs_per_recipient_per_block {
+                // Create outputs for each token id in round-robin fashion
+                for token_id in 0..=max_token_id {
+                    let amount = Amount {
+                        value: picomob_per_output,
+                        token_id: token_id.into(),
+                    };
+                    let output = TxOut::new(amount, recipient, &tx_private_key, Default::default());
+                    outputs.push(output.unwrap());
+                }
+            }
+        }
+
+        // Create key images unless we're at the origin block
+        let key_images: Vec<KeyImage> = if previous_block.is_some() {
+            (0..key_images_per_block)
+                .map(|_i| KeyImage::from(rng.next_u64()))
+                .collect()
+        } else {
+            Default::default()
+        };
+
+        let block_contents = BlockContents {
+            key_images,
+            outputs: outputs.clone(),
+            ..Default::default()
+        };
+
+        // Create a block with the desired contents
+        let block = match previous_block {
+            Some(parent) => {
+                Block::new_with_parent(block_version, &parent, &Default::default(), &block_contents)
+            }
+            None => Block::new_origin_block(&outputs),
+        };
+
+        previous_block = Some(block.clone());
+        blocks_and_contents.push((block, block_contents));
+    }
+    blocks_and_contents
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
     use mc_transaction_core::compute_block_id;
 
+    #[test]
+    // `get_custom_test_ledger_blocks` should return blocks that match the
+    // configuration specified in the arguments and pass all normal
+    // consistency tests
+    fn test_custom_block_correctness() {
+        let blocks_and_transactions = get_custom_test_ledger_blocks(2, 3, 3, 3, 0);
```

Comment:
Suggest storing the values in local variables so that the derived assertions are easier to associate.
also suggest avoiding repeated values so it's easier to ensure the correct input was used in the correct location.
i.e.
```
let outputs_per_recipient_per_block = 2;
let num_accounts = 3;
let num_blocks = 4;
let key_images_per_block = 5;
let max_token_id = 6;
let blocks_and_transactions = get_custom_test_ledger_blocks(outputs_per_recipient_per_block, num_accounts, num_blocks, key_images_per_block, max_token_id);
...
assert_eq!(block.cumulative_txo_count, (block.index + 1) * (outputs_per_recipient_per_block * num_blocks));
...
    assert_eq!(block_contents.key_images.len(), key_images_per_block);
```

Reply:
I think this would be overkill, in terms of the test, it's easy enough to look up at the parameter list in the file.